src/shared/XLinkDevice.c

```diff
@@ -479,6 +484,15 @@ static linkId_t getNextAvailableLinkUniqueId()
 }
 
 static xLinkDesc_t* getNextAvailableLink() {
+
+    XLINK_RET_ERR_IF(pthread_mutex_lock(&availableXLinksMutex) != 0, NULL);
```

Comment:
Heh, there's more and more need for c++, to use a unique lock and "forget" about handling all the branches the code can take :)